### PR TITLE
Add data facade utility

### PR DESCRIPTION
**Pull request by dgca**
Adds a `data-facade` package so that we can define the API interface and its implementation in isolation from the React app, so that we can swap out data sources in the future while reducing the need to refactor the React components themselves.

### CODE REVIEW

File: packages/data-facade/README.md [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+## Motivation
+
+The initial version of the mobile app will get its data from the `wallet-server`, a server that gets data from an Iron Fish node. Though the data source is fixed for now, we want to make it easy to switch to a different data source in the future. By using this abstraction layer, we'll be able to switch between data sources without needing to make changes to the React components themselves.
+
+## Usage
+
+1. Define your interface and create handlers
+2. Create facade context and export `FacadeProvider` and `useFacade`
+3. Wrap your application with `FacadeProvider`
+4. Use `useFacade` hook to access the facade in your components
```
Comment (dgca) [inline packages/data-facade/README.md:14]: I wanna expand on this, but figured I'd keep it simple while I get feedback on the implementation.

File: packages/mobile-app/data/accounts/handlers.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import { f } from "data-facade";
+import { z } from "zod";
+import { AccountsMethods } from "./types";
+
+const accounts = ["alice", "bob", "carol"];
+
```
Comment (jowparks) [inline packages/mobile-app/data/accounts/handlers.ts:6]: I am expecting that we have a handlers file per datasource? ie /accounts/mock.ts, /accounts/walletServer.ts, etc?

File: packages/mobile-app/data/index.ts [hunk 3]
```diff
@@ -0,0 +1,7 @@
+import { createFacadeContext } from "data-facade";
+import { accountsHandlers } from "./accounts/handlers";
+
+const facadeContext = createFacadeContext(accountsHandlers);
```
Comment (jowparks) [inline packages/mobile-app/data/index.ts:4]: This is importing the mock implementation, how will this look when we have multiple handlers for different systems? Will there be a conditional here based on a setting? I am imagining that a user might change data backend at any point. Will the backend change result in a account reset/chain reset? Will that have any impact on your representation here?

**Review by jowparks — APPROVED** [review]
Only one major comment about the different providers. Would be nice to have that sorted, but I don't think it is totally critical to do in this particular PR.

**Review by dguenther — APPROVED** [review]

**Pull request merged**